autoload/maktaba/log.vim

```diff
@@ -68,6 +75,31 @@ function! s:SendToHandlers(logitem) abort
 endfunction
 
 
+function! s:NotifyMessage(message, level) abort
+  if a:level >= s:LEVELS.ERROR
+    call maktaba#error#Shout(a:message)
+  elseif a:level >= s:LEVELS.WARN
+    call maktaba#error#Warn(a:message)
+  else
+    echomsg a:message
+  endif
+endfunction
+
+
+""
+" Sets the minimum {level} that will be immediately shown to the user using
```

Comment:
You don't mention `echomsg` here (for level < `WARN`), and I think perhaps it might be clearer to reorder slightly?

Perhaps something like the following?

```viml
""
" Sets the minimum {level} of log messages that will trigger a user
" notification, or -1 to disable notifications. By default, the user will be
" notified after every message logged at WARN or higher.
"
" Notifications will be sent using @function(maktaba#error#Shout) for ERROR
" and SEVERE messages, @function(maktaba#error#Warn) for WARN, and |:echomsg|
" for INFO and DEBUG.
" @throws BadValue
```